fvm/environment/programs.go

```diff
@@ -428,6 +171,96 @@ func (programs *Programs) cacheMiss() {
 	programs.metrics.RuntimeTransactionProgramsCacheMiss()
 }
 
+// programLoader is used to load a program from a location.
+type programLoader struct {
+	loadFunc        func() (*interpreter.Program, error)
+	dependencyStack *dependencyStack
+	called          bool
```

Comment:
is the programLoader used only for loading a single address?

Is it possible the same loader being used by loading multiple addresses?  In that case, the `called` just means it's called for loading the first address. 

Do we call `loadFunc` twice if loading the same address twice?

Reply:
loadFunc is being called only once. and is not reused in any way. Let me see if I can make this clearer in the code.

Reply:
Added a panic and a comment if called multiple times.

Reply:
(this comment is actually for https://github.com/onflow/flow-go/pull/4002 i just forgot to change the base)